litellm/caching.py

```diff
@@ -139,8 +152,23 @@ async def async_increment(self, key, value: float, **kwargs) -> float:
         init_value = await self.async_get_cache(key=key) or 0
         value = init_value + value
         await self.async_set_cache(key, value, **kwargs)
+
+        if time.time() - self.last_cleaned > self.default_ttl:
+            asyncio.create_task(self.clean_up_in_memory_cache())
+
         return value
 
+    async def clean_up_in_memory_cache(self):
+        """
+        Runs periodically to clean up the in-memory cache
+
+        - loop through all keys in cache, check if they are expired
+        - if yes, delete them
+        """
+        self.cache_dict = {}
```

Comment:
I am surprised you don't try to keep some entries around. Maybe I misunderstand the code, I imagined you'd need to keep entries for the current minute at least.

Reply:
fixed to use a custom eviction policy 

```python
def evict_cache(self):
        """
        Eviction policy:
        - check if any items in ttl_dict are expired -> remove them from ttl_dict and cache_dict
        This guarantees the following:
        - 1. When item ttl not set: At minimumm each item will remain in memory for 5 minutes
        - 2. When ttl is set: the item will remain in memory for at least that amount of time
        - 3. the size of in-memory cache is bounded
        """
        for key in list(self.ttl_dict.keys()):
            if time.time() > self.ttl_dict[key]:
                self.cache_dict.pop(key, None)
                self.ttl_dict.pop(key, None)
```